src/js/index.js

```diff
@@ -64,14 +87,19 @@ function handleTags() {
 	Array.from(contentEl.querySelectorAll('.item')).forEach(el => {
 		const show = (showExpired.checked || !el.classList.contains('tag-expired')) &&
 			tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), tags.length === 0);
-		if (!show) {
+
+		const hide = hideCompleted.checked && el.querySelector('.complete-notice').checked;
```

Comment:
I think this logic can be tied together. However, I think we should definitely comment that different cases that affect visibility